front/app/utils/errorUtils.ts

```diff
@@ -146,29 +147,78 @@ export function getDefaultAjvErrorMessage({
   );
 }
 
-export const handleHookFormSubmissionError = (
-  error: Error | CLErrorsJSON,
+// There's a similar function above but it's kind of insane, so
+// I'm creating a new one
+export const isCLErrorsJSON = (value: unknown): value is CLErrorsJSON => {
+  return isObject(value) && 'json' in value && isObject(value.json.errors);
+};
+
+// This is to check if it's not the JSON wrapper but the normal response
+export const isCLErrorsWrapper = (value: unknown): value is CLErrorsWrapper => {
+  return isObject(value) && isObject(value.errors);
+};
+
+// This one checks both. Needed because right now the 'old' utils/request
+// and the new fetcher deal with errors differently (the former wraps it in json)
```

Comment:
extra bracket in comment. Wonder if it's worth adding a TODO: here so that it gets flagged to remove at a later date?